### ISSUE TITLE
[Bug] Best-checkpoint saved during VQC training is not actually the best epoch's weights

### ISSUE THREAD

**Issue by inii-sayaa** [rasidi3112/Quantum-Machine-Learning#2]
## Description
In `train_variational_model` (qml_app/training.py), the "best model" checkpoint is
saved with:

    best_state_dict = model.state_dict()

`model.state_dict()` returns tensors that share the same underlying storage as the
live model parameters (it calls `.detach()`, not `.clone()`). Because training
continues after this line, subsequent `optimizer.step()` calls mutate those same
tensors in place — so `best_state_dict` silently changes along with the model,
even though it was "captured" at the epoch with the lowest val_loss.

## Evidence
Verified with a minimal repro:

    import torch, torch.nn as nn
    m = nn.Linear(2, 2)
    sd = m.state_dict()
    before = sd['weight'].clone()
    opt = torch.optim.SGD(m.parameters(), lr=1.0)
    loss = m(torch.ones(1, 2)).sum()
    loss.backward()
    opt.step()
    print(torch.equal(before, sd['weight']))  # -> False: sd changed after step()

## Impact
`model.load_state_dict(best_state_dict)` at the end of training (line ~130) loads
whatever the weights happened to be at that point (essentially the final epoch,
not the true best-val-loss epoch). This defeats early stopping / best-checkpoint
selection and quietly makes VQC evaluation numbers unreliable.

## Location
qml_app/training.py, line 121 (best_state_dict assignment) and line 130 (load).

## Suggested fix
    import copy
    best_state_dict = copy.deepcopy(model.state_dict())

## Severity
Critical - affects correctness of all reported VQC results.
